content/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/gulfstream-overview/index.md

```diff
@@ -0,0 +1,90 @@
+
+CRDs are only made up of data — which is often schematized — but they are brought to life by controller processes and webhooks that conventionally run on the cluster itself, serving as a sort of extension to the built-in control plane processes. 
+
+With traditional controller architectures, every cluster in a fleet runs every one of the clusters needed to run certain abstractions. This means that it's increasingly common for platform operations teams to be managing multiple extensions across many clusters, as in this diagram.
+
+![Diagram similar to the first in this guide, showing multiple clusters across three area zones (us-west2, us-central3, and us-east1). However, this diagram highlights the extensions running on each cluster.](clusters_per_AZ_extensions.png)
+
+As the number of clusters grows, so does the difficulty and cost of offering managed extensions to each of them, potentially growing into a complexity nightmare.
+
+
+## Understanding Controllers, Reconcilers, and Control Planes
+
+With the spread of CRDs, it can be difficult to know how trustworthy a given extension is. When you extend Kubernetes with extensions based on CRDs, you install additional control planes which implement the semantics of their extension. A *control plane* consists of the various APIs that control the behavior of systems; in Kubernetes, the API server and its controllers form the control plane. 
+
+A *controller* ensures that the actual state of a given object meets its desired state through a process called *reconciliation*. The part of the code that enforces reconciliation is called a *reconciler*. The controller watches various predefined resources; events occur on these resources, prompting reconciliation.
+
+Knative's controller architecture separates the logic of controllers and reconcilers, as the following diagram represents.
+
+![Diagram outlining a typical controller architecture. Within the controller are informers (each with a lister cache), a workqueue, clients, and a reconciler. The informers communicate with the workqueue, which passes events to the reconciler. The reconciler then communicates with the clients and lister cache as it performs reconciliation. Meanwhile, the informers and clients can communicate with an external API server.](controller_architecture.png)
+
+In this architecture, the controller is aware of informers and the workqueue, sets up the various relationships between event triggers and the workqueue, and hands off clients and listers to the reconciler. Reconcilers know nothing about informers, they just know about the clients and listers in order to access what they need to enable reconciliation.
+
+
+## Trusting trusted control planes
+
+[SLSA](https://slsa.dev/) makes several references to the idea of *trusted control planes*. Put simply, a trusted control plane is one that you trust in some way that you do not trust other control planes or workloads.
```

Comment:
As the next paragraph clarifies, there are reasons for there to be trust in "trusted control planes." Could we maybe combine the paragraphs and remove the "in some way" phrasing to make it clearer?

Reply:
I'm having a really hard time losing the "in some way" phrasing, so what I did was add an example borrowed from MM's blog post on trusted control planes. This will tie it back to SLSA a bit more and provide more context for the following "supervisor processes" paragraph. I think it's a lot clearer than what I originally had